webcompat/helpers.py

```diff
@@ -397,7 +398,18 @@ def wrapped_func(*args, **kwargs):
                 file_path = FIXTURES_PATH + request.path + "." + checksum
             else:
                 file_path = FIXTURES_PATH + request.path
-            if not os.path.exists(file_path + '.json'):
+            if app.config['UPDATE_MOCK_OUTS']:
+                data = func(*args, **kwargs)
+                with open(file_path + '.json', 'w') as f:
+                    data_json = json.loads(data[0])
+                    if isinstance(data_json, str):
+                        data_json[0]["_fixture"] = True
+                    if isinstance(data_json, dict):
+                        data_json["_fixture"] = True
+                    data_json = json.dumps(data_json, indent=2)
+                    f.write(data_json)
```

Comment:
Before trying to understand why UPDATE_MOCK_OUTS is failing, I want to understand what this code is trying to do. 

fixtures are preserved in the history of the project through a git commit in the project. Their parameters are carefully chosen to work and fail given a certain number of parameters. Sometimes they are like the real issue online, sometimes they are not. So it's important to not update them when running the app. They should be read but never written by the app. That would defeat the purpose of the fixture.

Loading them and saving them in a test environment will not change them in the git history. 

It seems that what you need is a tool to adjust some specific fixtures **once** then commit them in the project.

This should probably be a tooling script under `tools/` But I might have misunderstood what you are trying to do.

Reply:
>It seems that what you need is a tool to adjust some specific fixtures once then commit them in the project.

That's exactly what I want to do.

>I want to understand what this code is trying to do.

I needed to modify all of the mock outs we had to be able to remove the `markdown-it`. 
The way `markdown-it` works is the following: it receives a content formatted as markdown and then translates it to html. But, the new way I'm doing is making a request to Github where the content is already in html.
I can't pass the tests because the mock outs have this old markdown format and the program is expecting to deal with html now, so it crashes.
This code should enable the user to get into this "update mock outs mode" where the user can download the new version of the mock outs from the repository and it will be formatted and saved automatically.

>Sometimes they are like the real issue online, sometimes they are not. So it's important to not update them when running the app.

That's a problem. If the repo contains different versions of the issues we're using to make the tests, than I think it's useless.

>This should probably be a tooling script under `tools/`

I didn't know that. I asked Mike and he said it was okay to do that way, but I think I should probably have asked you :p
- sorry for that. If you think this is still useful I could try to rewrite this on `tools/`. But if you think it isn't... Then I could just do the easiest thing and substitute the old markdown content with html with a ctrl+c/ctrl_v

Reply:
>  This code should enable the user to get into this "update mock outs mode" where the user can download the new version of the mock outs from the repository and it will be formatted and saved automatically.
> Does this answer your question or would you like me to be more specific?

It does. So indeed this should be a script in tools as it is to manage the project and is not really part of the app. 

> That's a problem. If the repo contains different versions of the issues we're using to make the tests, than I think it's useless.

:) not necessary. That's the usual issue with mocking in general. Mocking is a way to isolate the tests from external factors (for example http requests to GitHub when testing). If GitHub is changing its API, some tests might still pass and we would not know. The bet/assumption we make when we are mocking is that the API is changing less often than when we run the tests.

That's the first part of mocking.

The second part of mocking is to be able to test again cases you want to fail. Sometimes you want to create a response that you are not likely to get but if you were getting it, you want to be able to handle it. Responses might evolve too. Think about HTTP headers  and things like pagination for example.

So indeed updating tests is often unrewarding. :) 

1. Make the tool you need in `tools/` to adjust the tests that's a good idea to include it, even if we run it only once every 6 months. Whatch out for fixtures which are not for functional tests. `nosetests` is **not** using the `python run.py -t` mode but is using some files in fixtures.
2. Commit the tool.
3. Run the tool to change the fixtures in your markdown-it removal branch (careful, not all fixtures are about functional tests)
4. Run the functional tests and see what is breaking. We might learn things.
5. Commit the new tests and the markdown-it removal in the same pull request.
6. If everything is working as expected, we should be able to merge :)

Thanks a lot @marimeireles 

> I didn't know that. I asked @miketaylr and he said it was okay to do that way, but I think I should probably have asked you :p

hehe. It's why it's often better to discuss options in the issues. So people can jump in with rationales for different options. :)

Reply:
>Whatch out for fixtures which are not for functional tests. nosetests is not using the python run.py -t mode but is using some files in fixtures.

Okay. Thanks for the checklist :)

>hehe. It's why it's often better to discuss options in the issues. So people can jump in with rationales for different options. :)

You're right. I'll learn from this mistake haha